bot/pagination.py

```diff
@@ -62,14 +73,26 @@ def add_line(self, line: str = '', *, empty: bool = False) -> None:
         """
         Adds a line to the current page.
 
-        If the line exceeds the `self.max_size` then an exception is raised.
+        If the line exceeds `self.max_size`, then `self.max_size` will go up to `scale_to_size` for
+        a single line before creating a new page. If it is still exceeded, the excess characters
+        are stored and placed on the next pages until there are none remaining (by word boundary).
+
+        Raises a RuntimeError if `self.max_size` is still exceeded after attempting to continue
+        onto the next page.
 
         This function overrides the `Paginator.add_line` from inside `discord.ext.commands`.
 
         It overrides in order to allow us to configure the maximum number of lines per page.
         """
-        if len(line) > self.max_size - len(self.prefix) - 2:
-            raise RuntimeError('Line exceeds maximum page size %s' % (self.max_size - len(self.prefix) - 2))
+        remaining_words = None
+        if len(line) > (max_chars := self.max_size - len(self.prefix) - 2):
+            if len(line) > self.scale_to_size:
+                line, remaining_words = self._split_remaining_words(line, max_chars)
+                # If line still exceeds scale_to_size, we were unable to split into a second
+                # page without truncating.
+                if len(line) > self.scale_to_size:
+                    raise RuntimeError(f'Line exceeds maximum scale_to_size {self.scale_to_size}'
+                                       ' and could not be split.')
```

Comment:
I'm somewhat undecided about whether it would be better to raise a `RuntimeError` or just truncate the remaining characters if `add_line()` contains a single word in `line` that contains more than `scale_to_size` characters. Since it would not occur from regular usage at all, I'm leaning towards an exception but I could definitely be persuaded otherwise.

Reply:
I think it should be truncated. I don't really see a benefit in raising an error. What if someone _really_ does want to print such a long string? There's no reason we should stop them. It'll also be easier for users of the paginator if they don't have to worry about such cases in the first place.

Reply:
Alright, I'll go with truncation then and update the docstring accordingly.

Reply:
I'm mostly just not a fan of truncating at specific point in the message and potentially cutting something off, but in the case of a single continuous word that exceeds `scale_to_size` it likely wouldn't cut off anything important.